examples/products.js

```diff
@@ -0,0 +1,155 @@
+             (fnode.element.getAttribute('title').includes(title.innerHTML) || title.innerHTML.includes(fnode.element.getAttribute('title'))) ||
+             fnode.element.getAttribute('alt') &&
+             (fnode.element.getAttribute('alt').includes(title.innerHTML) || title.innerHTML.includes(fnode.element.getAttribute('alt')))){
+               return 10000;
+             };
+    }
+
+    const rules = ruleset(
+      //get all images
+      rule(dom('img'), type('images')),
+
+      //better score for larger images
+      rule(type('images'), score(imageSize)),
+
+      //make sure image has src
+      rule(type('images'), score(imageHasSrc)),
+
+      //image title matches page title
+      rule(type('images'), score(imageTitle)),
+
+      //return image with max score
+      rule(type('images').max(), out('main-image'))
+
+    );
+
+    function contentFnodes(doc) {
+        return rules.against(doc).get('main-image');
+    }
+
+    return contentFnodes;
+}
+
+/**
+ * Maintain state as we compare a series of DOMs, reporting the percent
+ * difference at the end.
+ */
+class DiffStats {
+    constructor(contentFnodes) {
+        this.lengthOfExpectedTexts = 0;
+        this.lengthOfDiffs = 0;
+        this.contentFnodes = contentFnodes || tunedImageFnodes();
+    }
+
+    compare(expectedDom, sourceDom) {
+        //compare images by src
+        const expectedText = expectedDom.body.firstChild.getAttribute('src');
+        const gotText = this.contentFnodes(sourceDom).map(fnode => fnode.element.getAttribute('src'))[0];
+
+        this.lengthOfExpectedTexts += expectedText.length;
+        this.lengthOfDiffs += leven(expectedText, gotText);
+
+        console.log(leven(expectedText, gotText), expectedText.length, leven(expectedText, gotText)/expectedText.length);
+
+        // Uncomment for debugging:
+        console.log('Got:\n' + gotText);
+        console.log('\nExpected:\n' + expectedText);
+    }
+
+    score() {
+        return this.lengthOfDiffs / this.lengthOfExpectedTexts * 100;
+    }
+}
+
+
+function deviationScore(docPairs, coeffs = []) {
+    const stats = new DiffStats(tunedImageFnodes(...coeffs));
+    window.width = 1661;
+    window.height = 562;
+    for (let pair of docPairs) {
+        window.document.body.insertAdjacentHTML('afterbegin', '<div id = "fixture">' + pair[1] + '</div>');
+        if(document.getElementsByTagName('title')[1] !== undefined){
+          console.log(document.getElementsByTagName('title')[1].innerHTML);
+        }
+        stats.compare(pair[0], window.document);
+        window.document.body.removeChild(document.getElementById('fixture'));
+    }
+    console.log(stats.score());
+    return stats.score();
+}
+
+/** Return (expected DOM, source DOM) for all the readbaility test docs. */
+function productDocPairs() {
+  return [
+    [staticDom(fs.readFileSync(__dirname + '/../test/product_classification_test_data/amazon/expected-image.html', 'utf-8')),
```

Comment:
It sure is a lot of ugly noise, but I don't know what else you could have done that brfs could have followed.